spinalcordtoolbox/gui/base.py

```diff
@@ -0,0 +1,592 @@
+import sys
+import webbrowser
+from copy import copy
+from time import time
+import os
+
+import PyQt4.QtCore as QtCore
+import PyQt4.QtGui as QtGui
+import matplotlib as mpl
+import matplotlib.pyplot as plt
+import sct_utils as sct
+from matplotlib import cm
+from matplotlib.backends.backend_qt4agg import FigureCanvasQTAgg as FigureCanvas
+from matplotlib.lines import Line2D
+from msct_image import Image
+from msct_types import Coordinate
+from numpy import pad, percentile
+
+class ImagePlot(object):
+    """
+    Core class for displaying pictures.
+    Defines the data to display, image parameters, on_scroll event, manages intensity and manages the click (in or out the picture,...).
+    """
+    def __init__(self, ax, images, viewer, canvas, view, line_direction='hv', im_params=None, header=None):
+        self.axes = ax
+        self.images = images
+        self.viewer = viewer
+        self.view = view
+        self.line_direction = line_direction
+        self.image_dim = self.images[0].data.shape
+        self.figs = []
+        self.cross_to_display = None
+        self.aspect_ratio = None
+        self.canvas = canvas
+        self.last_update = time()
+        self.mean_intensity = []
+        self.std_intensity = []
+        self.list_intensites = []
+        self.im_params = im_params
+        self.current_position = Coordinate(
+            [int(self.images[0].data.shape[0] / 2), int(self.images[0].data.shape[1] / 2),
+             int(self.images[0].data.shape[2] / 2)])
+        self.list_points = []
+        self.header = header
+        self.dic_translate_label = self.define_translate_dic()
+
+        self.remove_axis_number()
+        self.connect_mpl_events()
+        self.setup_intensity()
+
+    def define_translate_dic(self):
+        """
+        Defines dictionnary to translate the software labels which range is [1;27] into anatomical labels which range is:
+        {50;49} U {1} U [3,26]
+
+        Returns
+        -------
+        dic        dictionnary
```

Comment:
I propose : 
Dictionnary that links the label position, from top to down, to the label definition according to https://github.com/neuropoly/spinalcordtoolbox/issues/1205